temporal/fx.go

```diff
@@ -258,7 +258,8 @@ func ServerOptionsProvider(opts []ServerOption) (serverOptionsProvider, error) {
 
 		esClient, err = esclient.NewClient(esConfig, esHttpClient, logger)
 		if err != nil {
-			return serverOptionsProvider{}, fmt.Errorf("unable to create Elasticsearch client: %w", err)
+			return serverOptionsProvider{}, fmt.Errorf("unable to create Elasticsearch client (URL = %v, username = %q): %w",
+				esConfig.URL, esConfig.Username, err)
```

Comment:
Is there any risk of leaking sensitive credentials this way?

Reply:
Hm. I think URL and username should be fine.
cc: @yiminc

Reply:
@bergundy URL does not hold a password (password is in esConfig) the URL.String() output will be in the following format, which seems safe to me:
`scheme://userinfo@host/path?query#fragment`

Reply:
oh actually, maybe not lm check again

Reply:
Updated it to `Redacted`